src/solid/lib/base.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  clearNode,
+  createNode,
+  createWatch,
+  is,
+  Scope,
+  scopeBind,
+  step,
+  Store,
+  Unit,
+} from 'effector'
+import {throwError} from './throw'
+import {Accessor, batch, createMemo, createSignal, onCleanup} from 'solid-js'
+
+const stateReader = <T>(store: Store<T>, scope?: Scope) =>
+  scope ? scope.getState(store) : store.getState()
+const basicUpdateFilter = <T>(upd: T, oldValue: T) => upd !== oldValue
+
+export function useUnitBase<Shape extends {[key: any]: Unit<any>}>(
+  shape: Shape,
+  scope?: Scope,
+) {
+  const isShape = !is.unit(shape) && typeof shape === 'object'
+  const normShape = isShape ? shape : {unit: shape}
+  const isList = Array.isArray(normShape)
+  const entries = Object.entries(normShape)
+
+  const initial = isList ? [] : {}
+  const stores = []
+  const storeIdMap = {}
+  const storeSetterMap = {}
+
+  for (const [key, value] of entries) {
+    if (is.store(value)) {
+      if (stores.includes(value)) {
+        throwError(`useUnit store at key "${key}" is already exists in shape`)
```

Comment:
Do we actually need to runtime-ban such cases? 🤔 

I can see usecases like this:
```ts
const useEllipseParams = (stores) => {
 return useUnit({
   leftFocalPoint: stores.leftFocalPoint,
   rightFocalPoint: stores.rightFocalPoint,
   ...
 })
}
```
where `left...` and `right...` can be the same store, if we want to draw a circle:
```ts
const circleParams = useEllipseParams({
 leftFocalPoint: $point,
 rightFocalPoint: $point
})
```
I think that in cases, where shape of `useUnit` argument have some meaning by itself, there sometimes can be the same store in different keys and we should not block it 🤔 

I guess, it should be more like this instead:

```ts
      const hasStore = Boolean(storeIdMap[value.graphite.id])
      
      if (!hasStore) {
        storeIdMap[value.graphite.id] = key
        const [get, set] = createSignal(stateReader(value, scope))
        initial[key] = get
        storeSetterMap[key] = set
      } else {
        const originalKey = storeIdMap[value.graphite.id]
        
        initial[key] = initial[originalKey]
        storeSetterMap[key] = storeSetterMap[originalKey] 
      }
```

Reply:
True, also makes sense to fix effector-react version as well then